From 911460d342ff681da7b22143ee74e652de342b3c Mon Sep 17 00:00:00 2001
From: Mihir Shete <smihir@qti.qualcomm.com>
Date: Tue, 11 Mar 2014 21:46:21 +0530
Subject: [PATCH] wlan: initialize temp_reg_domain from NV file.

If the temp_reg_domain is not initialzed and the first
regulatory hint comes directly from the CORE when wiphy
is registered then we might end up indexing the wrong
regdomain in the NV structure which leads to memory
corruption. To avoid this initialize temp_reg_domain
with the regulatory domain value obtained from NV
before wiphy initialization.

Change-Id: I6251d08b831af82f69da0ad562b5396bd9e30d8a
CRs-Fixed: 629647
---
 CORE/VOSS/src/vos_nvitem.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CORE/VOSS/src/vos_nvitem.c b/CORE/VOSS/src/vos_nvitem.c
index c768613..15f14eb 100644
--- a/CORE/VOSS/src/vos_nvitem.c
+++ b/CORE/VOSS/src/vos_nvitem.c
@@ -3682,6 +3682,8 @@ VOS_STATUS vos_init_wiphy_from_nv_bin(void)
         wiphy->flags |= WIPHY_FLAG_STRICT_REGULATORY;
     }
 
+    temp_reg_domain = cur_reg_domain = reg_domain;
+
     m = 0;
     for (i = 0; i < IEEE80211_NUM_BANDS; i++)
     {
-- 
1.8.2

